prime-router/src/main/kotlin/cli/tests/HistoryApiTest.kt

```diff
@@ -0,0 +1,233 @@
+    val httpStatus: Int,
+    val id: ReportId?,
+    val topic: String?,
+    val reportItemCount: Int?,
+    val warningCount: Int?,
+    val errorCount: Int?
+)
+
+data class SubmissionAPITestCase(
+    val name: String,
+    val path: String,
+    val headers: Map<String, String>,
+    val parameters: List<Pair<String, Any?>>?,
+    val bearer: String,
+    val expectedHttpStatus: HttpStatus,
+    val expectedReports: Set<ReportId>,
+)
+
+class HistoryApiTest : CoolTest() {
+    override val name = "history"
+    override val description = "Test the History/Lineage API"
+    override val status = TestStatus.SMOKE
+
+    // todo this code was copied from oktaAccessToken in SettingCommands.kt
+    // This is now copied in 3 places:  here, SettingCommands, LookupTableCommands, in 3 different ways.
+    // Best style is the lazy init in SettingCommands.  To factor up we need to pass Environment to
+    // CoolTest constructor.  This is a lotta work.
+    /**
+     * The access token left by a previous login command as specified by the command line parameters
+     */
+    fun getAccessToken(environment: Environment): String {
+        return if (environment.oktaApp == null) {
+            "dummy"
+        } else {
+            OktaCommand.fetchAccessToken(environment.oktaApp)
+                ?: abort(
+                    "Cannot run test $name.  Invalid access token. " +
+                        "Run ./prime login to fetch/refresh your access token for the $environment environment."
+                )
+        }
+    }
+
+    /**
+     * Create some fake history, so we have something to query for.
+     * @return null on failure.  Otherwise returns the list of ReportIds created.
+     */
+    private fun submitTestData(environment: Environment, options: CoolTestOptions): Set<ReportId>? {
+        val receivers = listOf<Receiver>(csvReceiver)
+        val counties = receivers.map { it.name }.joinToString(",")
+        val fakeItemCount = receivers.size * options.items
+        ugly("Starting $name test: Submitting ${options.submits} reports, each going to to $counties")
+        val file = FileUtilities.createFakeFile(
+            metadata,
+            settings,
+            historyTestSender,
+            fakeItemCount,
+            receivingStates,
+            counties,
+            options.dir,
+        )
+        echo("Created datafile $file")
+        // Now send it to ReportStream over and over
+        val reportIds = (1..options.submits).map {
+            val (responseCode, json) =
+                HttpUtilities.postReportFile(
+                    environment,
+                    file,
+                    historyTestSender,
+                    options.asyncProcessMode,
+                    options.key,
+                    payloadName = "$name ${status.description}",
+                )
+            echo("Response to POST: $responseCode")
+            if (responseCode != HttpURLConnection.HTTP_CREATED) {
```

Comment:
I wonder why sometimes we use HttpURLConnection (HTTPUtility) as HTTP Client.  Other times,  we use Fuel as HTTP Client?  Can use only Fuel?

Reply:
Yes, this is a legacy of different coders making different choices.  We should all use Fuel.